tests/admin_checks/tests.py

```diff
@@ -138,6 +142,88 @@ class SongAdmin(admin.ModelAdmin):
         ]
         self.assertEqual(errors, expected)
 
+    def test_wrong_prepopulated_fields(self):
+        class SongAdmin(admin.ModelAdmin):
+            prepopulated_fields = {'title': 'test'}
+
+        errors = SongAdmin(Song, AdminSite()).check()
+        expected = [
+            checks.Error(
+                'The value of \'prepopulated_fields["title"]\' must be a list or tuple.',
+                obj=SongAdmin,
+                id='admin.E029',
+            )
+        ]
+        self.assertEqual(errors, expected)
+
+    def test_wrong_readonly_fields(self):
+        class SongAdmin(admin.ModelAdmin):
+            readonly_fields = 'test'
+
+        errors = SongAdmin(Song, AdminSite()).check()
+        expected = [
+            checks.Error(
+                "The value of 'readonly_fields' must be a list or tuple.",
+                obj=SongAdmin,
+                id='admin.E034',
+            )
+        ]
+        self.assertEqual(errors, expected)
+
+    def test_wrong_list_editable(self):
+        class SongAdmin(admin.ModelAdmin):
+            list_editable = 'test'
+
+        errors = SongAdmin(Song, AdminSite()).check()
+        expected = [
+            checks.Error(
+                "The value of 'list_editable' must be a list or tuple.",
+                obj=SongAdmin,
+                id='admin.E120',
+            )
+        ]
+        self.assertEqual(errors, expected)
+
+    def test_list_editable_missing_field(self):
+        class SongAdmin(admin.ModelAdmin):
+            list_editable = ('test',)
+
+        errors = SongAdmin(Song, AdminSite()).check()
+        expected = [
+            checks.Error(
+                "The value of 'list_editable[0]' refers to 'test', which is not an attribute of 'admin_checks.Song'.",
```

Comment:
These longer strings can be wrapped around 79 chars as done in some other tests.